pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -298,12 +298,59 @@ type SecuritySpec struct {
 	// +optional
 	// +nullable
 	KeyManagementService KeyManagementServiceSpec `json:"kms,omitempty"`
-	// KeyRotation defines options for Key Rotation.
+	// KeyRotation defines options for rotation of OSD disk encryption keys.
 	// +optional
 	// +nullable
 	KeyRotation KeyRotationSpec `json:"keyRotation,omitempty"`
 }
 
+// ClusterSecuritySpec is the CephCluster security spec to include various security items such as kms
+type ClusterSecuritySpec struct {
+	// KeyManagementService is the main Key Management option
+	// +optional
+	// +nullable
+	KeyManagementService KeyManagementServiceSpec `json:"kms,omitempty"`
+	// KeyRotation defines options for rotation of OSD disk encryption keys.
+	// +optional
+	// +nullable
+	KeyRotation KeyRotationSpec `json:"keyRotation,omitempty"`
```

Comment:
Instead of brining in these two properties directly, should the `SecuritySpec` be referenced, similar to the ObjectStoreSecuritySpec?
```
SecuritySpec `json:""`
```

Reply:
I did that in a prior iteration. I decided to try this instead since the other way results in more complicated unit test updates across the many files. It can be done either way.